aws/retry/middleware.go

```diff
@@ -247,6 +253,32 @@ func (r *Attempt) handleAttempt(
 	return out, attemptResult, releaseRetryToken, err
 }
 
+// Note that there are errors that are known to be definitely caused by clock
+// skew, which are defined on the list of retryable errors
+var possibleSkewCodes = map[string]struct{}{
+	"InvalidSignatureException": {},
+	"SignatureDoesNotMatch":     {},
+	"AuthFailure":               {},
```

Comment:
Are `RequestTimeTooSkewed`, `RequestInTheFuture` and `RequestExpired` (added below) omitted here intentionally?

Reply:
Yes, they were added to the list of `definitely retry always`. Moved to a separate list as discussed on comment below